### PR TITLE
Add logging for Parser, Ui, DeleteCommand, ListCommand, and FlashcardList

### LINKED ISSUE #50: Add logging to Parser

### LINKED ISSUE #51: Add logging to Ui

### LINKED ISSUE #60: Add logging to DeleteCommand and ListCommand

### LINKED ISSUE #61: Add logging to FlashcardList

### PR DESCRIPTION
**Pull request by Zhilin-Huang**
First attempt at logging so I'm not exactly sure if this is the right way to add logs. Let me know what you guys think!

### CODE REVIEW

File: src/main/java/seedu/tp/commands/DeleteCommand.java [hunk 1]
```diff
@@ -15,10 +21,20 @@
      * @param index         index in the delete command
      */
     public DeleteCommand(FlashcardList flashcardList, int index) {
+        setupLogger();
         this.flashcardList = flashcardList;
         this.index = index;
     }
 
+    private static void setupLogger() {
```
Comment (quinnyyy) [inline src/main/java/seedu/tp/commands/DeleteCommand.java]: Is there no way we can do this with inheritance?

File: src/main/java/seedu/tp/flashcard/FlashcardList.java [hunk 2]
```diff
@@ -5,21 +5,34 @@
...
+        this.flashcards = new ArrayList<>();
     }
 
+    /**
+     * Constructor for FlashcardList.
+     *
+     * @param flashcardList the list of flashcards to be added
+     */
     public FlashcardList(List<Flashcard> flashcardList) {
+        setupLogger();
```
Comment (quinnyyy) [inline src/main/java/seedu/tp/flashcard/FlashcardList.java]: We can replace line 35 and 36 here with a call to this()

File: src/main/java/seedu/tp/flashcard/FlashcardList.java [hunk 3]
```diff
@@ -31,12 +44,22 @@ public FlashcardList(List<Flashcard> flashcardList) {
      * @param flashcardList the flashcard list to be copied from
      */
     public FlashcardList(FlashcardList flashcardList) {
+        setupLogger();
```
Comment (quinnyyy) [inline src/main/java/seedu/tp/flashcard/FlashcardList.java]: Same here

**Review by quinnyyy — COMMENTED** [review]
The main thing I am wondering about is if we can use inheritance to get rid of the code reuse in the Command class loggers. My idea is if we make a non-static logger member in the base class and move the setupLogger() method to the default constructor of the base class. I think we should definitely consider changing this if we are planning to add logging to the other Commands

File: src/main/java/seedu/tp/commands/DeleteCommand.java (hunk 1 above)
Comment (g-lilian) [reply src/main/java/seedu/tp/commands/DeleteCommand.java]: Agreed, it makes sense to put this in the base Command class.

File: src/main/java/seedu/tp/commands/DeleteCommand.java [hunk 4]
```diff
@@ -15,10 +21,20 @@
      * @param index         index in the delete command
      */
     public DeleteCommand(FlashcardList flashcardList, int index) {
+        setupLogger();
```
Comment (quinnyyy) [inline src/main/java/seedu/tp/commands/DeleteCommand.java]: I think that setupLogger needs to be called once when the program starts up, not every time we instnatiate a Command. logger is a static member so redoing the setup every time we create a new flashcard is wasteful. I think maybe we should have a method that runs at the start of the program that does set up of all the loggers. 

https://www.vogella.com/tutorials/Logging/article.html

We can see from this example that prof linked that they are calling MyLogger.setup() from main() in the last example code. What do u think?

File: src/main/java/seedu/tp/commands/DeleteCommand.java (hunk 1 above)
Comment (quinnyyy) [reply src/main/java/seedu/tp/commands/DeleteCommand.java]: I'm actually not sure if its a good idea anymore b/c I found this article that says it's a bad practice:
https://rolf-engelhard.de/2013/10/logging-anti-patterns-part-iii/
Although it doesn't really explain what's wrong with it

Comment (quinnyyy) [reply src/main/java/seedu/tp/commands/DeleteCommand.java]: actually nvm again i think its still a good idea lol

Comment (Zhilin-Huang) [reply src/main/java/seedu/tp/commands/DeleteCommand.java]: Thanks for the comments! I'll move the `setupLogger ()` to the base Command class.

File: src/main/java/seedu/tp/commands/DeleteCommand.java (hunk 4 above)
Comment (Zhilin-Huang) [reply src/main/java/seedu/tp/commands/DeleteCommand.java]: Agree :) I'll refactor the code to follow this example you shared. Thanks!

File: src/main/java/seedu/tp/flashcard/FlashcardList.java (hunk 2 above)
Comment (Zhilin-Huang) [reply src/main/java/seedu/tp/flashcard/FlashcardList.java]: Updated :)

File: src/main/java/seedu/tp/ui/Ui.java [hunk 5]
```diff
@@ -291,24 +328,26 @@ public void printException(Exception exception) {
...
         System.out.println("The flashcard index you entered is invalid");
     }
-    
+
     public void sendLoggingSetupFailedMessage() {
         System.out.println("Logging setup failed! Logs will be printed to console instead of saved to file.");
     }
 
     public void sendInvalidInputFormatResponse() {
         System.out.println("Please use the correct input format. Use \"help\" to view all commands.");
```
Comment (quinnyyy) [inline src/main/java/seedu/tp/ui/Ui.java:350]: Maybe add a logging statement here?

File: src/main/java/seedu/tp/ui/Ui.java [hunk 6]
```diff
@@ -291,24 +328,26 @@ public void printException(Exception exception) {
...
     }
-    
+
     public void sendDuplicateFlashcardResponse() {
```
Comment (quinnyyy) [inline src/main/java/seedu/tp/ui/Ui.java:353]: and here?

**Review by quinnyyy — COMMENTED** [review]
Maybe can add a couple more logging statements to the Ui, otherwise lgtm

**Comment by Zhilin-Huang** [comment]
> Maybe can add a couple more logging statements to the Ui, otherwise lgtm

Thanks for the feedback! Updated :)

**Review by quinnyyy — APPROVED** [review]

**Pull request merged**